packages/@jsii/kernel/lib/kernel.ts

```diff
@@ -140,8 +140,11 @@ export class Kernel {
     req: api.InvokeScriptRequest,
   ): api.InvokeScriptResponse {
     const result = cp.spawnSync(
-      path.join(this._getPackageDir(req.pkgname), req.script),
-      req.args,
+      'node',
+      [path.join(this._getPackageDir(req.pkgname), req.script)].concat(
+        req.args ?? [],
+      ),
```

Comment:
It's probably going to be more efficient to use the same `node` runtime that is used by the Kernel process (and it's also going to be safer in Windows):
- `process.execPath` contains the path to the `node` binary that runs the current process.
- `process.execArgv` contains the `node` arguments that the current process received.

The `--` indicates the end of `node` options and guarantees that none of your own command options are possibly interpreted by node itself.

```suggestion
      process.execPath,
      [
        ...process.execArgv,
        '--',
        path.join(this._getPackageDir(req.pkgname), req.script),
        ...req.args ?? [],
      ],
```